### PR TITLE
Return an empty item if altitude is not available from iOS GPS

### PR DESCRIPTION
**Pull request by livecodefraser**

### CODE REVIEW

File: engine/src/exec-sensor.cpp [hunk 1]
```diff
@@ -193,8 +193,43 @@ void MCSensorGetLocationOfDevice(MCExecContext& ctxt, MCStringRef &r_location)
...
     if (MCSystemGetLocationReading(t_reading, false))
-        // PM-2014-10-09: [[ Bug 12142 ]] The old %Lf format worked for device but failed on simulator.
-        MCStringFormat(r_location, "%lf,%lf,%lf", t_reading.latitude, t_reading.longitude, t_reading.altitude);
+    {
+        bool t_success = true;
+        MCAutoStringRef t_location(kMCEmptyString);
+        t_success = t_location.MakeMutable();
+        if (t_success)
+        {
+            if (!isnan(t_reading.latitude))
```
Comment (peter-b) [inline engine/src/exec-sensor.cpp]: I suggest using `isfinite()` rather than `isnan()` here (and generally).

**Comment by peter-b** [comment]
@livecode-vulcan review ok dca32b8

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok dca32b865ea4ee693453859f2a11ac881948e5d6

**Comment by livecode-vulcan** [comment]
😞 test failure dca32b865ea4ee693453859f2a11ac881948e5d6
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2603)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2613)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2598)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2614)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2620)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2626)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2638)

**Comment by peter-b** [comment]
@livecode-vulcan review ok f60b673

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok f60b673f55887cfb017f56525b2ad1a2b82fb80b

**Comment by livecode-vulcan** [comment]
😎 test success f60b673f55887cfb017f56525b2ad1a2b82fb80b
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2611)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2621)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2606)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2622)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2628)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2634)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2646)

**Pull request merged**